junit5-extension/src/main/java/io/kroxylicious/testing/kafka/junit5ext/TopicPartitions.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.testing.kafka.junit5ext;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import io.kroxylicious.testing.kafka.api.KafkaClusterConstraint;
+
+/**
+ * Used to specify the number of partitions of a {@link Topic} injected into
+ * a test as field or parameter.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.FIELD, ElementType.PARAMETER })
+@KafkaClusterConstraint
+public @interface TopicPartitions {
```

Comment:
I don't have strong views on this but I'd like to hear your thoughts on why partition and replica counts are separate annotations rather than properties of the topic annotation. 

The later feels like a slightly more natural expression to me (probably using nullable fields for inheriting broker defaults)

Reply:
> I don't have strong views on this but I'd like to hear your thoughts on why partition and replica counts are separate annotations rather than properties of the topic annotation.

My motivation was the cardinality - TopicConfig naturally needs to be repeatable in order to support > 1.

Reply:
A repeatable topicConfig makes sense but i meant  partitions and replication factor specifically

Reply:
Ah, so you mean introduce a `@Topic` annotation which would carry the replicationFactor/replicationFactor fields.  I presume that actually adding the annotation would remain optional.

So it would look like:

```
@Topic(replicationFactor = 2) Topic myTopic 
```

this form would work too.

```
Topic myTopic 
```

Reply:
Alternatively,  we could chose to special case the config names `replication.factor` and `partitions`, and use those to pass the configuration.  Those names aren't understood by the Kafka Client, but they do have currency already in Kafka Connect (`org.apache.kafka.connect.runtime.TopicCreationConfig#REPLICATION_FACTOR_CONFIG`).  That would avoid the need for the additional annotation but would mean we have the extra javadoc.

Reply:
> `@Topic(replicationFactor = 2) Topic myTopic `

Was exactly what I had in mind. I confess `@Topic Topic` does look a little odd. 

I think your right about `@TopicParitions` vs using `@TopicConfig` and handling repetition so the dedicated annotations make more sense than re-using `TopicConfig`.

*My personal* preference would still be an optional `@Topic` but I'm happy to merge this and come revisit later.

Reply:
I have a marginal preference for `@TopicParitions/`@TopicReplicationFactor` over `@Topic Topic` so I'm going to stick with my way.